src/com/twitter/intellij/pants/execution/PantsMakeBeforeRun.java

```diff
@@ -285,29 +293,57 @@ public void onTextAvailable(ProcessEvent event, Key outputType) {
     runningPantsProcesses.remove(currentProject, process);
 
     final boolean success = process.exitValue() == 0;
-    // Mark project dirty if compile failed.
-    if (!success) {
-      FileChangeTracker.markDirty(currentProject);
-    }
-    else {
-      FileChangeTracker.addManifestJarIntoSnapshot(currentProject);
+    if (tasks.contains(PantsConstants.PANTS_TASK_COMPILE)) {
+      if (success) {
+        // FIXME: is manifest jar always created? what is that? IS it always, if there's a compile?
```

Comment:
it is always created if the run succeeds.

Reply:
Thanks, noted.